tcpproxy.go

```diff
@@ -80,13 +80,8 @@ type Proxy struct {
 	ListenFunc func(net, laddr string) (net.Listener, error)
 }
 
-type route struct {
-	matcher matcher
-	target  Target
-}
-
-type matcher interface {
-	match(*bufio.Reader) bool
+type route interface {
```

Comment:
docs.

notably, match can't consume bytes from bufio.Reader, and it can return nil to not match.